File: backend/FinTrack/FinTrack.Domain/Expenses/ExpenseCategory.cs
namespace FinTrack.Domain.Expenses;

public enum ExpenseCategory
{
    Rent = 1,
    Utilities = 2,
    Groceries = 3,
    Transportation = 4,
    Insurance = 5,
    Healthcare = 6,
    Education = 7,
    Entertainment = 8,
    DiningOut = 9,
    Shopping = 10,

    // Subcategories
    MedicalBills = 11,
    PrescriptionDrugs = 12,
    GymMembership = 13,
    StreamingServices = 14,
    Subscriptions = 15,

    // Work-related expenses
    OfficeSupplies = 16,
    BusinessTravel = 17,
    ProfessionalDevelopment = 18,

    // Debt-related categories
    LoanRepayment = 19,
    CreditCardPayment = 20,
    Savings = 21,

    // Miscellaneous categories
    Gifts = 22,
    CharitableDonations = 23,
    Taxes = 24,
    EmergencyFund = 25
}

File: frontend/fintrack_app/lib/features/expenses/domain/enums/expense_category.dart
enum ExpenseCategory {
  rent(1),
  utilities(2),
  groceries(3),
  transportation(4),
  insurance(5),
  healthcare(6),
  education(7),
  entertainment(8),
  diningOut(9),
  shopping(10),

  // Subcategories
  medicalBills(11),
  prescriptionDrugs(12),
  gymMembership(13),
  streamingServices(14),
  subscriptions(15),

  // Work-related expenses
  officeSupplies(16),
  businessTravel(17),
  professionalDevelopment(18),

  // Debt-related categories
  loanRepayment(19),
  creditCardPayment(20),
  savings(21),

  // Miscellaneous categories
  gifts(22),
  charitableDonations(23),
  taxes(24),
  emergencyFund(25);

  final int value;
  const ExpenseCategory(this.value);
}
